qiskit_experiments/library/characterization/analysis/tphi_analysis.py

```diff
@@ -26,6 +29,22 @@ class TphiAnalysis(CompositeAnalysis):
     A class to analyze :math:`T_\phi` experiments.
     """
 
+    def __init__(self, analyses=None):
+        if analyses is None:
+            analyses = [T1Analysis(), T2RamseyAnalysis()]
+
+        # Validate analyses kwarg
+        if (
+            len(analyses) != 2
+            or not isinstance(analyses[0], T1Analysis)
```

Comment:
User should be able to use custom analysis directly inheriting from `BaseAnalysis`. So only length check should be performed.

Reply:
For Tphi analysis? Isn't this one supposed to be hardcoded? The only reason I added the `analyses` kwarg was I could think of a nice way of setting the analyses classes to be references to those from the component T1 and T2 experiments without setting `._analyses = [expt1.analysis, expt2.analysis]` directly in the Tphi experiment init.

Reply:
Umm... probably you're right. This is how the experiment is originally implemented and outside the scope of this PR.